storage/decorator.py

```diff
@@ -412,12 +412,15 @@ def __call__(
                             self._unfinished_line = b""
 
             # in this case we could have a trailing new line in what's left in the buffer
-            # or the content had a leading `{` but was not a json object before the circuit breaker intercepted it:
+            # or the content had a leading `{` but was not a json object before the circuit breaker intercepted it
+            # or we waited for the object start and never reached:
             # let's fallback to by_lines()
             if not self._is_a_json_object:
-                for line, starting_offset, ending_offset, original_newline_length, _ in self._by_lines_fallback(
-                    self._unfinished_line
-                ):
+                buffer: bytes = self._unfinished_line
```

Comment:
the changes in this file address the third bugfix:
- proper handling of non json content where no json object start was ever met in the storage json collector

if we reached this if branch, the content was not identified as a json content, information hold by the flag `self._is_a_json_object`. This flag value start as `False`, and will be set to `True` only if at least one json object was able to be parsed from the storage content, that's correct

the if branch yields then `self._unfinished_line`, where all the previous storage content that was collected before the circuit breaker kicked in

the bug resides in the fact that that previous storage content is collected on `self._unfinished_line`, only if something that looked like a potential json object was ever met (ie: a line having a leading `{`).
we still have the case where this scenario never happened: this information is hold on the `wait_for_object_start` flag (where `False` means we finally found a potential json object) and the previous storage content that was collected so fare is instead in `wait_for_object_start_buffer`